builtin/providers/aws/resource_aws_launch_configuration.go

```diff
@@ -206,3 +461,9 @@ func resourceAwsLaunchConfigurationDelete(d *schema.ResourceData, meta interface
 
 	return nil
 }
+
+func readBlockDevicesFromLaunchConfiguration(d *schema.ResourceData, launchConfiguration *autoscaling.LaunchConfiguration, autoscalingconn *autoscaling.AutoScaling) map[string]interface{} {
+	var blockDevices = make(map[string]interface{})
+	// Need to figure out how to determine the various instance types.
```

Comment:
Okay, this is super rough and probably won't compile, I think it could look something like this: https://gist.github.com/phinze/01f6163bf452eee289d5

Basically, checking the `DeviceName` against the `RootDeviceName` of the AMI, leaning on a helper living over in `resource_aws_instance.go` to grab it for us.